webapp/graphite/events/views.py

```diff
@@ -34,9 +34,17 @@ def view_events(request):
 
 
 def detail(request, event_id):
-    e = get_object_or_404(Event, pk=event_id)
-    context = {'event': e}
-    return render_to_response("event.html", context)
+    if request.META['HTTP_ACCEPT'] == 'application/json':
+        e = Event.objects.get(id=event_id)
```

Comment:
The 404 case isn't covered for JSON requests. It'd be nice to return a 404 JSON response if `Event.DoesNotExist` is raised.